### PR TITLE
ESP32 C3 RX support

### PR DESCRIPTION
**Pull request by pkendall64**
Since the ESP8285 is aging and only has 2 more years of support from Espressif, the C3 is a good replacement.
The C3 has a few more GPIOs, does not do stupid things with to the IO pins during boot and has more flash/RAM etc.

The C3 also allows updates via USB (GPIO18/19) so if these pins are not used for anything then the firmware can be updated via direct USB connection. If the pins are used then the BOOT pin (GPIO 9) needs to be held low on power up. The C3 can also be updated via UART on GPIOs 20/21 as per other ESP32/ESP8285 chips, also by pulling GPIO9 low on power up.

The following is my janky setup for initial testing.
IMG_0667

And connected to my PWM testing bench.
IMG_0668

### CODE REVIEW

**Comment by pkendall64** [comment]
For those playing along at home, this is the C3 dev board in the pictures https://www.aliexpress.com/item/1005005791323624.html

**Comment by pkendall64** [comment]
> Can it be used as the main control chip of the RC internal TX? Replaces the original ESP32.

Technically, yes. But the dev team will NOT be doing that. The reason is that there are not enough GPIO pins to support all the features required for a TX module that would pass the approval process.

**Review by JyeSmith — APPROVED** [review]

**Comment by L1cardo** [comment]
Wow! I have been waiting for this for a long time! 
Finally here.
Thank you @pkendall64 !

**Review by wvarty — APPROVED** [review]

**Comment by pkendall64** [comment]
I'll merge this after the 3.x.x merge to master. There may be conflicts.

**Pull request merged**
